extensions-api/src/main/scala/pl/touk/nussknacker/engine/api/ProcessListener.scala

```diff
@@ -8,6 +8,8 @@ trait ProcessListener extends Lifecycle {
 
   def nodeEntered(nodeId: String, context: Context, processMetaData: MetaData): Unit
 
+  def nodeExited(nodeId: String, nextNodeId: Option[String], context: Context, processMetaData: MetaData): Unit
```

Comment:
Let's look for some not enter/exit -biased name. I think transition is often used in such cases. But the lone "transition" looks strange, we need some "event" / "context" / ? prefix